packages/video_player/tizen/src/video_player.cc

```diff
@@ -49,22 +49,37 @@ static std::string StateToString(player_state_e state) {
   return ret;
 }
 
+bool VideoPlayer::IsValidMediaPacket(media_packet_h media_packet) {
+  tbm_surface_h surface;
+  int ret = media_packet_get_tbm_surface(media_packet, &surface);
+  if (ret != MEDIA_PACKET_ERROR_NONE) {
+    LOG_ERROR("get tbm surface failed, error: %d", ret);
+    return false;
+  }
+  return true;
+}
+
 FlutterDesktopGpuBuffer *VideoPlayer::ObtainGpuBuffer(size_t width,
                                                       size_t height) {
   std::lock_guard<std::mutex> lock(mutex_);
-  if (media_packet_ == nullptr) {
-    LOG_ERROR("Media packet  is null");
-    return nullptr;
+  if (prepared_media_packet_ && !IsValidMediaPacket(prepared_media_packet_)) {
+    media_packet_destroy(prepared_media_packet_);
+    prepared_media_packet_ = nullptr;
   }
-  tbm_surface_h surface;
-  int ret = media_packet_get_tbm_surface(media_packet_, &surface);
-  if (ret != MEDIA_PACKET_ERROR_NONE) {
-    LOG_ERROR("media_packet_get_tbm_surface failed, error: %d", ret);
-    media_packet_destroy(media_packet_);
-    media_packet_ = nullptr;
+  if (current_media_packet_ && !IsValidMediaPacket(current_media_packet_)) {
+    media_packet_destroy(current_media_packet_);
+    current_media_packet_ = nullptr;
+  }
+  if (!prepared_media_packet_ && !current_media_packet_) {
+    LOG_ERROR("No vaild media packet");
     return nullptr;
   }
-
+  if (prepared_media_packet_ && !current_media_packet_) {
+    current_media_packet_ = prepared_media_packet_;
```

Comment:
Maybe we need a lock here? I'm a little worried.

Reply:
we add lock when entering the frame decode callback.